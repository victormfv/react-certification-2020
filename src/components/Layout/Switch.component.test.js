import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Switch from './Switch.component';

describe('Switch Component', () => {

  it('renders correctly', () => {
    const tree = renderer.create(<Switch text="Dark Mode" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Title is correct', () => {
    render(<Switch text="Dark Mode" />);
    const i = screen.getByText(/Dark Mode/i);
    expect(i).toBeInTheDocument();
  });
});