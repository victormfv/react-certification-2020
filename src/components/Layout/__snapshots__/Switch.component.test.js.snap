// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component renders correctly 1`] = `
<div>
  <label
    className="sc-bdfBwQ hfvcRx"
  >
    <input
      defaultChecked={true}
      type="checkbox"
    />
    <span
      className="sc-gsTCUz knPZT"
    />
  </label>
   
  Dark Mode
</div>
`;
